### PR TITLE
fix: 修复隐藏列总计时行总计也被隐藏问题

### PR DESCRIPTION
**Pull request by wuhaiyang**
### 👀 PR includes

✨ Feature

- [ ] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [x] 修复隐藏列总计时行总计也被隐藏问题

🔧 Chore

- [ ] Test case
- [ ] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 📝 Description

- 解决：目前仅支持隐藏列头部分，行总计和列总计ID 一致时也会被隐藏问题
- 其他说明：行头、列头实际场景不会出现同样的维度

### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
| iShot_2023-11-16_14 40 39       |  iShot_2023-11-16_14 39 27    |

### 🔍 Self-Check before the merge

- [ ] Add or update relevant docs.
- [ ] Add or update relevant demos.
- [ ] Add or update test case.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

**Comment by wuhaiyang** [comment]
已补充

File: packages/s2-core/__tests__/spreadsheet/hidden-columns-spec.ts [hunk 1]
```diff
@@ -554,6 +554,39 @@ describe('SpreadSheet Hidden Columns Tests', () => {
         expect(leafNodes.some((node) => node.id === nodeId)).toBeFalsy();
         expect(leafNodes).toHaveLength(5);
       });
+
+      test('hiding the column totals should not hide the row totals', () => {
```
Comment (lijinke666) [inline packages/s2-core/__tests__/spreadsheet/hidden-columns-spec.ts:558]: 两个测试断言一样用 `test.each` 即可

```suggestion
      test.each(['grid', 'tree'])('hiding the column totals should not hide the row totals for %s mode', (hierarchyType) => {
      
        sheet.setOptions({ hierarchyType);
        sheet.render(false)
```

Comment (wuhaiyang) [reply packages/s2-core/__tests__/spreadsheet/hidden-columns-spec.ts]: done

**Review by lijinke666 — APPROVED** [review]

**Pull request merged**